python/lsst/meas/modelfit/cmodel/cmodel.cc

```diff
@@ -0,0 +1,235 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#include "pybind11/pybind11.h"
+
+#include "numpy/arrayobject.h"
+#include "ndarray/pybind11.h"
+
+#include "lsst/pex/config/pybind11.h"
+#include "lsst/meas/modelfit/CModel.h"
+
+namespace py = pybind11;
+using namespace pybind11::literals;
+
+namespace lsst {
+namespace meas {
+namespace modelfit {
+namespace {
+
+using PyCModelStageControl = py::class_<CModelStageControl, std::shared_ptr<CModelStageControl>>;
+using PyCModelControl = py::class_<CModelControl, std::shared_ptr<CModelControl>>;
+using PyCModelStageResult = py::class_<CModelStageResult, std::shared_ptr<CModelStageResult>>;
+using PyCModelResult = py::class_<CModelResult, std::shared_ptr<CModelResult>>;
+using PyCModelAlgorithm = py::class_<CModelAlgorithm, std::shared_ptr<CModelAlgorithm>>;
+
+PyCModelStageControl declareCModelStageControl(py::module & mod) {
+    PyCModelStageControl cls(mod, "CModelStageControl");
+    cls.def(py::init<>());
+    cls.def("getProfile", &CModelStageControl::getProfile);
+    cls.def("getModel", &CModelStageControl::getModel);
+    cls.def("getPrior", &CModelStageControl::getPrior);
+    LSST_DECLARE_CONTROL_FIELD(cls, CModelStageControl, profileName);
+    LSST_DECLARE_CONTROL_FIELD(cls, CModelStageControl, priorSource);
+    LSST_DECLARE_CONTROL_FIELD(cls, CModelStageControl, priorName);
+    LSST_DECLARE_NESTED_CONTROL_FIELD(cls, CModelStageControl, linearPriorConfig);
+    LSST_DECLARE_NESTED_CONTROL_FIELD(cls, CModelStageControl, empiricalPriorConfig);
+    LSST_DECLARE_CONTROL_FIELD(cls, CModelStageControl, nComponents);
+    LSST_DECLARE_CONTROL_FIELD(cls, CModelStageControl, maxRadius);
+    LSST_DECLARE_CONTROL_FIELD(cls, CModelStageControl, usePixelWeights);
+    LSST_DECLARE_NESTED_CONTROL_FIELD(cls, CModelStageControl, optimizer);
+    LSST_DECLARE_CONTROL_FIELD(cls, CModelStageControl, doRecordHistory);
+    LSST_DECLARE_CONTROL_FIELD(cls, CModelStageControl, doRecordTime);
+    return cls;
+}
+
+PyCModelControl declareCModelControl(py::module & mod) {
+    PyCModelControl cls(mod, "CModelControl");
+    cls.def(py::init<>());
+    LSST_DECLARE_CONTROL_FIELD(cls, CModelControl, psfName);
+    LSST_DECLARE_NESTED_CONTROL_FIELD(cls, CModelControl, region);
+    LSST_DECLARE_NESTED_CONTROL_FIELD(cls, CModelControl, initial);
+    LSST_DECLARE_NESTED_CONTROL_FIELD(cls, CModelControl, exp);
+    LSST_DECLARE_NESTED_CONTROL_FIELD(cls, CModelControl, dev);
+    LSST_DECLARE_CONTROL_FIELD(cls, CModelControl, minInitialRadius);
+    LSST_DECLARE_CONTROL_FIELD(cls, CModelControl, fallbackInitialMomentsPsfFactor);
+    return cls;
+}
+
+// Custom wrapper for views to std::bitset.
+template <int N>
+class BitSetView {
```

Comment:
pybind11 also has lacks support for `std::bitset`. so wrapping the `flags` attributes directly wasn't an option.  I could have proceeded with the same workaround I had for Swig (the `getFlag`/`setFlag` methods), but I preferred the `.flags[...]` approach, and now that it was relatively easy to do that in Python (though not trivial), I decided to delete the old one.  That was before I realized how much code (including C++ code) was using what I considered the Swig-workaround interface, but I still think it's an improvement to have only one way to get and set these flags and to have it look similar across languages.